### DISCUSSION TITLE
How to select a marker programmaticaly?

### DISCUSSION THREAD

**Discussion by maciejpos** [rorystephenson/flutter_map_marker_popup#50] [General]
How to select a marker programmatically? I can't see the corresponding method in [PopupController](https://pub.dev/documentation/flutter_map_marker_popup/latest/flutter_map_marker_popup.extension_api/PopupController-class.html)

**Comment by rorystephenson** [comment]
Hi @maciejpos . If you want to select a single marker use: https://pub.dev/documentation/flutter_map_marker_popup/latest/flutter_map_marker_popup.extension_api/PopupController/showPopupsOnlyFor.html

**Reply by maciejpos** [reply]
Thank you @rorystephenson. However, I'm trying to select a marker (_without_ showing a popup). Let's assume you have 4 markers and you would like to programmatically select 3 of them and show a popup only for the 4th one (without selecting it). Is it possible?

**Reply by rorystephenson** [reply]
Sorry I let this question slip. Right now this is not supported and it feels a little bit out of the scope of this plugin. "Selected" markers was probably a bad name choice by my as it is supposed to mean the popups for which a popup is visible. So by that logic it wouldn't make much sense to support selecting a marker to show a popup and then not actually showing the popup.

At the same time I've run in to a similar use case myself and I ended up enhancing `flutter_map_supercluster` to make it possible. I can go in to further detail but it's not really relevant if you're not using that plugin. It would also be possible to support such behaviour with a custom marker layer and I can elaborate on that if you need.
